$NetBSD: patch-aa,v 1.7 2007/02/07 17:15:24 tron Exp $

Fix for GCC 3.3.x build problem taken from the SVN repository:

http://anonsvn.wireshark.org/viewvc/viewvc.py/trunk/tools/lemon/lemon.c?r1=20452&r2=20737&pathrev=20737&view=patch

Details of the problem are document here:

http://bugs.wireshark.org/bugzilla/show_bug.cgi?id=1357

--- tools/lemon/lemon.c	2007/01/16 16:41:12	20452
+++ tools/lemon/lemon.c	2007/02/07 16:08:19	20737
@@ -3247,7 +3247,10 @@
   for(i=0; i<rp->nrhs; i++) used[i] = 0;
   lhsused = 0;
 
-  if (!rp->code) rp->code = "\n";
+  if (!rp->code) {
+	  rp->code = "\n";
+	  rp->line = rp->ruleline;
+  }
   
   append_str(0,0,0,0);
   for(cp=rp->code; *cp; cp++){
